openbas-api/src/main/java/io/openbas/rest/inject/InjectApi.java

```diff
@@ -64,511 +66,554 @@
+          Duration.between(injectStatus.getTrackingSentDate(), trackingEndDate).getSeconds());
+      if (injectStatus.getTraces().stream()
+          .filter(injectStatusExecution -> injectStatusExecution.getStatus().equals(ExecutionStatus.ERROR)).count()
+          >= injectStatus.getTrackingTotalCount()) {
+        injectStatus.setName(ExecutionStatus.ERROR);
+      } else if (injectStatus.getTraces().stream().anyMatch(trace -> trace.getStatus().equals(ExecutionStatus.ERROR))) {
+        injectStatus.setName(ExecutionStatus.PARTIAL);
+      } else if (injectStatus.getTraces().stream()
+          .filter(injectStatusExecution -> injectStatusExecution.getStatus().equals(ExecutionStatus.MAYBE_PREVENTED))
+          .count() >= injectStatus.getTrackingTotalCount()) {
+        injectStatus.setName(ExecutionStatus.MAYBE_PREVENTED);
+      } else if (injectStatus.getTraces().stream()
+          .anyMatch(trace -> trace.getStatus().equals(ExecutionStatus.MAYBE_PREVENTED))) {
+        injectStatus.setName(ExecutionStatus.MAYBE_PARTIAL_PREVENTED);
+      } else {
+        injectStatus.setName(ExecutionStatus.SUCCESS);
+      }
     }
-
-    @Secured(ROLE_ADMIN)
-    @PostMapping(INJECT_URI + "/execution/callback/{injectId}")
-    public Inject injectExecutionCallback(@PathVariable String injectId, @Valid @RequestBody InjectExecutionInput input) {
-        Inject inject = injectRepository.findById(injectId).orElseThrow(ElementNotFoundException::new);
-        InjectStatus injectStatus = inject.getStatus().orElseThrow(ElementNotFoundException::new);
-        ExecutionStatus executionStatus = ExecutionStatus.valueOf(input.getStatus());
-        InjectStatusExecution execution = new InjectStatusExecution();
-        Instant trackingEndDate = now();
-        execution.setTime(trackingEndDate);
-        execution.setStatus(executionStatus);
-        execution.setMessage(input.getMessage());
-        execution.setIdentifiers(input.getIdentifiers());
-        injectStatus.getTraces().add(execution);
-        if (executionStatus.equals(ExecutionStatus.SUCCESS)) {
-            injectStatus.setTrackingTotalSuccess(injectStatus.getTrackingTotalSuccess() + 1);
-        } else {
-            injectStatus.setTrackingTotalError(injectStatus.getTrackingTotalSuccess() + 1);
+    return injectRepository.save(inject);
+  }
+
+
+  @GetMapping("/api/injects/try/{injectId}")
+  public Inject tryInject(@PathVariable String injectId) {
+    return atomicTestingService.tryInject(injectId);
+  }
+
+  @GetMapping(INJECT_URI + "/test/{injectId}")
+  public InjectStatus testInject(@PathVariable String injectId) {
+    Inject inject = injectRepository.findById(injectId).orElseThrow();
+    User user = this.userRepository.findById(currentUser().getId()).orElseThrow();
+    List<ExecutionContext> userInjectContexts = List.of(
+        this.executionContextService.executionContext(user, inject, "Direct test")
+    );
+    io.openbas.execution.Injector executor = context.getBean(
```

Comment:
io.openbas.execution.Injector should be replaced by the imported package
Injector